devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java

```diff
@@ -325,79 +326,107 @@ public void execute() throws MojoExecutionException {
         }
     }
 
-    static ExtensionCatalog resolveExtensionsCatalog(String groupId, String artifactId, String version,
+    static ExtensionCatalog resolveExtensionsCatalog(AbstractMojo mojo, String groupId, String artifactId, String version,
             ExtensionCatalogResolver catalogResolver, MavenArtifactResolver artifactResolver, MessageWriter log)
             throws MojoExecutionException {
 
         if (!catalogResolver.hasRegistries()) {
-            // TODO: this should normally result in an error, however for the time being
-            // until we get the registry service up and running this will allow
-            // a fall back to the legacy way of resolving the default platform catalog directly
-            return ToolsUtils.resolvePlatformDescriptorDirectly(groupId, artifactId,
-                    version == null ? CreateUtils.resolvePluginInfo(CreateUtils.class).getVersion() : version, artifactResolver,
-                    log);
+            groupId = getPlatformGroupId(mojo, groupId);
+            artifactId = getPlatformArtifactId(artifactId);
+            version = getPlatformVersion(mojo, version);
+            final StringBuilder buf = new StringBuilder();
+            buf.append("The extension catalog will be narrowed to the ").append(groupId).append(":").append(artifactId)
+                    .append(":").append(version).append(" platform release.");
+            buf.append(
+                    " To enable the complete Quarkiverse extension catalog along with the latest recommended platform releases, please, make sure ");
+            if (QuarkusProjectHelper.isRegistryClientEnabled()) {
+                buf.append("the following registries are accissible from your environment: ");
```

Comment:
```suggestion
                buf.append("one of the following registries are accissible from your environment: ");
```

I assume it just needs one to be present ?

Reply:
Actually at this point it will fail if at least one isn't available. It would have to be disabled or removed from the config.